src/cljc/ataru/hakija/application_validators.cljc

```diff
@@ -172,16 +172,40 @@
     (every? string? value) value
     (every? map? value) (map :value value)))
 
+(defn offending-priorization [hakukohde-oid selected priorisoivat-hakukohderyhmat]
+  (let [priorisoivat     (->> priorisoivat-hakukohderyhmat
+                              (mapcat (fn [o] (filter #(contains? (set %) hakukohde-oid) (:prioriteetit o))))
+                              (map (fn [o] (partition-by #(= hakukohde-oid %) o))))
```

Comment:
`partition-by`:ssa on se ongelma, että siinä syntyy erikoistapaus jos etsittävä elementti on ensimmäinen tai viimenen. Jos tekis näin
```
(->> priorisoivat-hakukohderyhmat
     (map (fn [ryhma]
            (map set (:prioriteetit ryhma))))
     (map (fn [prioriteetit]
            (cond (contains? (first prioriteetit) hakukohde-oid)
                  [#{} (set (rest prioriteetit))]
                  (contains? (last prioriteetit) hakukohde-oid)
                  [(set (butlast prioriteetit)) #{}]
                  :else
                  (let [[higher this lower] (partition-by #(contains? % hakukohde-oid)
                                                          prioriteetit)]
                    [(apply clojure.set/union higher) (apply clojure.set/union lower)])))))
```
niin sit pitäs saada lista pareja `[#{A B} #{C D}]` jotka tarkottaa, että A ja B pitää olla ennen hakukohdetta `hakukohde-oid` ja C ja D jälkeen.